### DISCUSSION TITLE
How would I "idiomatically" extend the deno root pattern set in the default nvim-lspconfig?

### DISCUSSION THREAD

**Discussion by luckenco** [LazyVim/LazyVim#2295] [Q&A] [answered]
When working with supabase only a `import_map.json` is used. How would I optimally extend the `root_pattern` for denols?

**Comment by dpetka2001** [comment, marked as answer]
Theoritically you could do something like 
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        denols = {
          root_dir = require("lspconfig.util").root_pattern("deno.json", "deno.jsonc", ".git", "import_map.json"),
        },
      },
    },
  },
```
Pls be aware that the order in the `root_dir` matters, since `lspconfig` will return the first successfully matched entry. So, change it according to which entry you would like to be matched first.

**Reply by luckenco** [reply]
thank you! this did the job for now. I will try to extend the patters instead of redefining them when play around it with my config the next time.
